app/scripts/services/tile-proxy.js

```diff
@@ -533,43 +528,45 @@ export const tileDataToPixData = (
 
 function text(url, callback) {
   /**
-   * Send a JSON request mark it so that we can tell how many are in flight
+   * Send a text request and mark it so that we can tell how many are in flight
    */
-  requestsInFlight += 1;
-  pubSub.publish('requestSent', url);
-
-  return fetch(url)
-    .then(rep => rep.text())
-    .then((text) => {
-      callback(undefined, text);
-    })
-    .catch((error) => {
-      callback(error, undefined);
-    })
-    .finally(() => {
-      pubSub.publish('requestReceived', url);
-      requestsInFlight -= 1;
-    });
+  fetchEither(url, callback, 'text');
 }
 
 function json(url, callback) {
   /**
-   * Send a JSON request mark it so that we can tell how many are in flight
+   * Send a JSON request and mark it so that we can tell how many are in flight
+   */
+  fetchEither(url, callback, 'json');
+}
+
+function fetchEither(url, callback, textOrJson) {
+  /**
+   * Send a either a text or JSON request and mark it so that we can tell how many are in flight
    */
   requestsInFlight += 1;
   pubSub.publish('requestSent', url);
 
-  const r = request(url)
-    .header('Content-Type', 'application/json')
-  // TODO: Check if this preserves same-origin cookies
-
-  if (authHeader)
-    r.header('Authorization', `${authHeader}`)
+  if (textOrJson == 'text') {
```

Comment:
Not crucial but in the JS world always use `===` instead of `==`. Often they are equivalent but sometimes funky things can happen, so it's recommended to just stick to `===`.